### PR TITLE
Productionize

### PR DESCRIPTION
**Pull request by btalberg**
You should be able to run this as a library, module, or CLI. Note the `.env.template` file, you'll need to cut a new `.env` and update some honeycomb vars in there. Also, you'll need to use this updated [ground_truth.csv](https://github.com/WildflowerSchools/wf-process-cuwb-data/files/5286257/ground_truth.csv.gz) file to generate training data.

To build the CLI, execute `just build` (You may need to install [just](https://github.com/casey/just) ). Then, to run the cli:

```
process_cuwb_data generate-tray-carry-groundtruth \
    --environment capucine \
    --start 2020-01-17T08:00:00-0500 \
    --end 2020-01-17T17:00:00-0500 \
    --groundtruth-csv <<ground_truth.csv path>>

process_cuwb_data train-tray-carry-model \
    --groundtruth-features <<groundtruth training file generated from generate-tray-carry-groundtruth>>

process_cuwb_data infer-tray-carry \
    --environment capucine \
    --start 2020-01-17T08:00:00-0500 \
    --end 2020-01-17T17:00:00-0500 \
    --model <<model generated from train-tray-carry-model>> \
    --feature-scaler <<feature scaler generated from train-tray-carry-model>>
```

FYI, I didn't touch the visualize.py or geom_render.py functions.

### CODE REVIEW

**Comment by tcquinn** [comment]
OK. I'm going to try to capture some of my questions here and then maybe we can chat before you head off. Important note: all questions below are true questions and not passive aggressive reactions! I'm still very much a novice programmer, both in general and in Python.

**Comment by tcquinn** [comment]
I would have thought that deleting all of those `import` statements from the top-level `__init__.py` would change the API in the sense that the library functions in the files other than `core.py` would no longer be available at the top-level name space (e.g., one can no longer reference `process_cuwb_data.plot_positions_multiple_devices()` Was that intentional?

**Comment by tcquinn** [comment]
In the top-level `__init__.py`, I've always opted for using the full path rather than the "dot" notation (i.e., I write `from process_cuwb_data.core import *` rather than `from .core import *` because it's more explicit and less fragile in the sense that it doesn't depend on the relative positions of the two files in the directory structure. However, I suppose my way is also _more_ fragile in the sense that, if I rename the package,  I have to propagate that renaming to lots of places. How do you think about this?

**Comment by tcquinn** [comment]
I've never used `just` but I gather it's a variation on `make`, which I never use either (for Python projects). What's the advantage of adding this extra layer on top of `python setup.py ...`? Does it give you more flexibility to specify different install options?

**Comment by tcquinn** [comment]
Inside my modules, when I'm importing other modules, I almost never use the `from ... import ...` pattern because it saves typing in my code but it decreases the explicitness of the code (it's not obvious from a quick glance where a particular function is coming from) and increases the risk of name space collisions. How do you think about this?

**Comment by tcquinn** [comment]
What is the `log.py` stuff doing for you? Is it just because you want a particular log format and/or destination than the default?

**Comment by tcquinn** [comment]
I think I just don't understand the command line interface stuff at all. The pattern(s) are totally different from what I know. If we have time, I'd love for you to just walk me through it.

**Comment by tcquinn** [comment]
I don't understand the PEP stuff, though I'm intrigued. Would love for you to walk me through it.

**Comment by tcquinn** [comment]
I don't understand how this stuff works. Would love to understand.
```
    @property
    def classifier(self):
        if self.__classifier is None:
            self.__classifier = RandomForestClassifier(
                **self.attrs()
            )
        return self.__classifier
```

**Comment by tcquinn** [comment]
I don't understand what the `dotenv` stuff is doing. The only things I see in the sample `.env` file are the Honeycomb credentials, but my Honeycomb libraries already access those. Would love to understand better.

**Review by tcquinn — APPROVED** [review]
Lots of questions, but it looks good to me. Feel free to merge before or after we touch base.
